src/OnlineSales/Data/ApiDbContext.cs

```diff
@@ -94,16 +94,18 @@ public override async Task<int> SaveChangesAsync(bool acceptAllChangesOnSuccess,
         {
             if (entityEntry.State == EntityState.Modified)
             {
-                ((BaseEntity)entityEntry.Entity).UpdatedAt = DateTime.UtcNow;
-                ((BaseEntity)entityEntry.Entity).UpdatedByIp = httpContextHelper!.IpAddress;
-                ((BaseEntity)entityEntry.Entity).UpdatedByUserAgent = httpContextHelper!.UserAgent;
+                var entity = (BaseEntity)entityEntry.Entity;
+                entity.UpdatedAt = entity.UpdatedAt is null ? DateTime.UtcNow : entity.UpdatedAt;
+                entity.UpdatedByIp = string.IsNullOrEmpty(entity.UpdatedByIp) ? httpContextHelper!.IpAddress : entity.UpdatedByIp;
+                entity.UpdatedByUserAgent = string.IsNullOrEmpty(entity.UpdatedByUserAgent) ? httpContextHelper!.UserAgent : entity.UpdatedByUserAgent;
```

Comment:
@gsplakmal same for user agent - it looks like user agent will always stay unchanged starting from second update and onwards. We need a test to verify this logic as well.